### PR TITLE
[CAPI] Update description of ml_train_datagen_cb

### PR DESCRIPTION
**Pull request by zhoonit**
- [CAPI] Update description of ml_train_datagen_cb

```
This patch updates ml_train_datagen_cb with newly added api sets and
clarify the parameter @a last means.

**Self evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test: [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: Jihoon Lee <jhoon.it.lee@samsung.com>

```

### CODE REVIEW

**Review by jijoongmoon — APPROVED** [review]
LGTM

File: api/nntrainer-api-common.h [hunk 1]
```diff
@@ -95,13 +95,16 @@ typedef enum {
  * number of samples are set as a property for this dataset. In this case, last
  * is only used for verification purposes. If total number of samples for the
  * dataset is unknown, this function will be called in sequence.
+ * @note If @a last false when not modified by the user. @a last has to be set
```
Comment (kparichay) [inline api/nntrainer-api-common.h:98]: If @a last false when not modified by the user. -> it sounds like `last` variable will be defaulted to false by the caller and user need not set it if not needed. Is this intentional?

Comment (zhoonit) [reply api/nntrainer-api-common.h:98]: Yes, it is current behavior and I think it's fine to behave like this.

Comment (kparichay) [reply api/nntrainer-api-common.h:98]: The described behavior works, but IMO, its best for the callee to always set this variable, and not assume about the default value set by the caller.

**Review by kparichay — APPROVED** [review]
LGTM

File: api/nntrainer-api-common.h (hunk 1 above)
Comment (zhoonit) [reply api/nntrainer-api-common.h:98]: Let's delete this line although this behavior will exist to not break old codes.

**Pull request merged**